### ISSUE TITLE
feature regression: struct packing zero length throws exception

### ISSUE THREAD

**Issue by ryannathans** [micropython/micropython#2114]
Python3.5
```
>>> import struct
>>> struct.pack("0s", bytearray(0))
b''
```

Micropython
```
>>> import struct
>>> struct.pack("0s", bytearray(0))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: unsupported format
```

**Comment by ryannathans** [comment]
This used to work in previous versions, I'm not yet sure which commit broke it.

```
MicroPython v1.6-246-gb894551-dirty on 2016-03-23; HydraBus1.0 with STM32F4
>>> import struct
>>> struct.pack("0s", b"")
b''
>>> struct.pack("0s", bytearray(0))
b''
```

**Comment by ryannathans** [comment]
This bug was introduced by commit e53fb1bf031a53ad23a45846addf14e25cebb97d

**Comment by dpgeorge** [comment]
Should be fixed by 715ee9d925871f2c44632442ff42cb58c99ee9c7.
